test/inferenceql/query/cache_test.cljc

```diff
@@ -0,0 +1,21 @@
+(ns inferenceql.query.cache-test
+  (:require [clojure.test :refer [deftest is testing]]
+            [inferenceql.query.cache :as cache]))
+
+(deftest basic-caching
+  (let [cache-size 2
+        a (atom 0)
+        incrementer (fn [_ignored-but-cached-key]
+                      (swap! a inc))
+        cached-incrementer (cache/lru incrementer cache-size)]
+
+    (is (= 1 (cached-incrementer :foo)))
```

Comment:
how about a test for cache awareness of `0` vs `nil`?

Reply:
Good point, especially since it's an area of concern

Reply:
done